stdlib/public/stubs/Stubs.cpp

```diff
@@ -265,16 +265,19 @@ swift_stdlib_readLine_stdin(unsigned char **LinePtr) {
 #endif
 }
 
-#if defined(__CYGWIN__) || defined(_WIN32)
-  #define strcasecmp _stricmp
-#endif
-
 static bool swift_stringIsSignalingNaN(const char *nptr) {
   if (nptr[0] == '+' || nptr[0] == '-') {
     ++nptr;
   }
 
-  return strcasecmp(nptr, "snan") == 0;
+  if ((nptr[0] == 's' || nptr[0] == 'S') &&
+      (nptr[1] == 'n' || nptr[1] == 'N') &&
+      (nptr[2] == 'a' || nptr[2] == 'A') &&
+      (nptr[3] == 'n' || nptr[3] == 'N')) {
```

Comment:
This needs to check index 4 for NUL to be equivalent. Not sure what our actual requirements are here.

Reply:
Right, this is supposed to preserve existing behavior. Thanks for pointing that out!